packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -1531,6 +1548,28 @@ export class Table extends TableBase {
     }));
   }
 
+  private setTimeToLive(timeToLiveAttribute?: string) {
+    const provider = TableTimeToLiveProvider.getOrCreate(this);
+
+    // Permissions
+    const permissions = ['dynamodb:DescribeTable', 'dynamodb:DescribeTimeToLive', 'dynamodb:UpdateTimeToLive'];
+    this.grant(provider.onEventHandler, ...permissions);
+    this.grant( provider.isCompleteHandler, ...permissions);
+
+    const timeToLive = new CustomResource(this, 'TimeToLive', {
+      serviceToken: provider.provider.serviceToken,
+      resourceType: 'Custom::DynamoDBTimeToLive',
+      properties: {
+        TableName: this.tableName,
+        TimeToLiveAttribute: timeToLiveAttribute !== '' ? timeToLiveAttribute : undefined,
+      },
+    });
+
+    timeToLive.node.addDependency(
+      this,
```

Comment:
The custom resource already references the table so there's no need to a dependency.

Reply:
You’re right, I will remove this.